com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Procedural/Noise/NoiseHashNode.cs

```diff
@@ -0,0 +1,120 @@
+                    },
+                    new string[]
+                    {
+                        "\"Packages/com.unity.render-pipelines.core/ShaderLibrary/Hashes.hlsl\""
+                    }
+                ),
+                new (
+                    "Hash22",
+                    @"    Hash_Tchou_2_2_float(UV*Scale, Out);",
+                    new ParameterDescriptor[]
+                    {
+                        new ParameterDescriptor("UV", TYPE.Vec2, Usage.In, REF.UV0),
+                        new ParameterDescriptor("Scale", TYPE.Vec2, Usage.In, new float[] {10.0f, 10.0f}),
+                        new ParameterDescriptor("Out", TYPE.Vec2, Usage.Out),
+                    },
+                    new string[]
+                    {
+                        "\"Packages/com.unity.render-pipelines.core/ShaderLibrary/Hashes.hlsl\""
+                    }
+                ),
+                new (
+                    "Hash31",
+                    @"    Hash_Tchou_3_1_float(Position*Scale, Out);",
+                    new ParameterDescriptor[]
+                    {
+                        new ParameterDescriptor("Position", TYPE.Vec3, Usage.In, REF.WorldSpace_Position),
+                        new ParameterDescriptor("Scale", TYPE.Vec3, Usage.In, new float[] {10.0f, 10.0f, 10.0f}),
+                        new ParameterDescriptor("Out", TYPE.Float, Usage.Out),
+                    },
+                    new string[]
+                    {
+                        "\"Packages/com.unity.render-pipelines.core/ShaderLibrary/Hashes.hlsl\""
+                    }
+                ),
+                new(
+                    "Hash33",
+                    @"    Hash_Tchou_3_3_float(Position*Scale, Out);",
+                    new ParameterDescriptor[]
+                    {
+                        new ParameterDescriptor("Position", TYPE.Vec3, Usage.In, REF.WorldSpace_Position),
+                        new ParameterDescriptor("Scale", TYPE.Vec3, Usage.In, new float[] {10.0f, 10.0f, 10.0f}),
+                        new ParameterDescriptor("Out", TYPE.Vec3, Usage.Out)
+                    }
+                )
+            }
+        );
+        public static NodeUIDescriptor NodeUIDescriptor => new(
+            Version,
+            Name,
+            tooltip: "generates a random value based on the input",
+            category: "Procedural/Noise",
+            displayName: "Noise Hash",
+            synonyms: new string[0] {  },
+            selectableFunctions: new()
+            {
+                { "Hash21", "Vec2 to Float" },
+                { "Hash23", "Vec2 to Vec3" },
+                { "Hash22", "Vec2 to Vec2" },
+                { "Hash31", "Vec3 to Float" },
+                { "Hash33", "Vec3 to Vec3" }
+            },
+            functionSelectorLabel: "Hash Type",
+            parameters: new ParameterUIDescriptor[4] {
+                new ParameterUIDescriptor(
+                    name: "UV",
```

Comment:
Love the name change! the graph issue of the wire getting lost occurs when switching the dropdown options tho, hope it will get fixed soon